### PR TITLE
[storage] index and serve TableInfo

### PR DESCRIPTION
**Pull request by msmouse**
### Description

Separate rocksdb in case we need to delete it for space on validator.

Didn't make it optional though for simplicity, I don't expect a lot of tables being created for now. and parsing the events shouldn't be a performance concern.

### Test Plan
Updated table UT in API to assert type info of test tables.

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2158)

### CODE REVIEW

File: aptos-move/framework/aptos-stdlib/sources/table.move [hunk 1]
```diff
@@ -96,6 +106,41 @@ module aptos_std::table {
...
+        assert!(!exists<TableMetadata>(@aptos_std), error::already_exists(ETABLE_METADATA_ALREADY_EXISTS));
+        move_to(account, TableMetadata {
+            new_table_events: new_event_handle<NewTableEvent>(account)
+        });
+    }
+
+    /// Emits NewTableEvent if the event handle is installed under @aptos_std.
+    fun maybe_emit_create_table_event<K: copy + drop, V>(handle: u128) acquires TableMetadata {
+        // In unit tests the event handle is not installed.
+        if (exists<TableMetadata>(@aptos_std)) {
```
Comment (davidiw) [inline aptos-move/framework/aptos-stdlib/sources/table.move:131]: why not

```
#[test_only]
return;
```

**Review by davidiw — COMMENTED** [review]
why another PR?

File: aptos-move/framework/aptos-stdlib/sources/table.move (hunk 1 above)
Comment (msmouse) [reply aptos-move/framework/aptos-stdlib/sources/table.move:131]: Because it doesn't work like that?

File: api/src/tests/state_test.rs [hunk 2]
```diff
@@ -255,6 +261,11 @@ async fn assert_table_item<T: Serialize, U: Serialize>(
     key: T,
     value: U,
 ) {
-    let response = api_get_table_item(ctx, table, key_type, value_type, key).await;
+    let handle = TableHandle(table["handle"].as_str().unwrap().parse().unwrap());
+    let table_info = ctx.db.get_table_info(handle).unwrap();
+    assert_eq!(table_info.key_type, parse_type_tag(key_type).unwrap());
+    assert_eq!(table_info.value_type, parse_type_tag(value_type).unwrap());
+
+    let response = api_get_table_item(ctx, handle.0, key_type, value_type, key).await;
```
Comment (msmouse) [inline api/src/tests/state_test.rs:269]: pointer

**Comment by msmouse** [comment]
closing in favor of https://github.com/aptos-labs/aptos-core/pull/2191

**Pull request closed without merging**